spec/index.html

```diff
@@ -2383,12 +2381,155 @@ <h4>YamlLdErrorCode</h4>
           </section>
         </section>
 
+        <section>
+          <h2>Implementations</h2>
+
+          <p>
+            TODO: Implementations for Extended Internal Representation.
+          </p>
+        </section>
+
       </section>
 
       <section>
         <h2>Convert Extended YAML-LD to Basic YAML-LD and back</h2>
 
-        TODO
+        <p>
+          This approach is simpler than the Extended Internal Representation because it
+          does not require any changes to the internal structures of existing JSON-LD
+          libraries.
+        </p>
+
+        <p>
+          Instead, we implement two API functions:
+        </p>
+
+        <dl data-sort>
+          <dt><code>extended_to_basic(extended_document: YAML-LD) → YAML-LD</code></dt>
+          <dd>
+            <ul>
+              <li>Performs JSON-LD expansion to the given Extended YAML-LD document</li>
```

Comment:
I still don't understand what JSON-LD is doing in a YAML-LD-to-YAML-LD transformation.

Reply:
It's necessary to be sure that the input document is in expanded form so that we don't need a mechanism parallel to JSON-LD expansion to properly handle all the term types, containers, and scoped contexts. Doing this in expanded form prevents what would otherwise be a very complicated process, in the general case.

Reply:
Something like this explanation should be put in the document, including how and why "JSON-LD expansion" is applied to a YAML-LD document, which isn't JSON-LD, which makes it very confusing that "JSON-LD expansion" is applied instead of "YAML-LD expansion".

Reply:
Hmm, this is puzzling. I see the following two alternatives how we can word this.

1. We just specify that source YAML-LD document has to be expanded beforehand.
   * but then, we have to specify what YAML-LD expansion is,
   * and the interface of the functions becomes not entirely user friendly: user has to _remember_ that the document has to be expanded, and the functions probably should _verify_ that.

2. Do the expansion ourselves:
   * Convert YAML → JSON,
   * Perform JSON-LD Expansion,
   * Convert resulting JSON back to YAML,
   * Perform the necessary basic → extended or extended → basic conversion.

I would lean to the option (2).

Of course, internally, when we're using `rdflib` for instance, — we won't actually convert YAML to JSON and back; instead we will read YAML into an internal Python data structure of nested dicts, lists, strings, numbers and `None`s, which `rdflib` will subsequently process. But I doubt we can say it that way in the spec.

Reply:
I think it's fine to say something like "You won't typically need to perform these steps manually, because libraries such as `rdflib` will take care of them under the covers, but it can help with troubleshooting and optimization to know what's going on. So, you start with YAML, convert it to JSON, perform JSON-LD Expansion, convert that to YAML-LD, and do any necessary basic → extended or extended → basic conversion on the YAML-LD. Alternatively, your library might do YAML-LD expansion directly on the initial YAML document, and then do any necessary basic → extended or extended → basic conversion on the YAML-LD."

Reply:
Thanks @TallTed! I added this paragraph.

Also, I think we actually can't convert Extended → Basic exactly through JSON-LD because Extended YAML-LD specifically can't be convert to JSON-LD. Right?

Reply:
You'll have to convert YAML to the YAML JSON profile, which is compatible.

Reply:
What about like this? Should extended → basic require conversion?

How do scoped contexts and stuff influence the conversion of type tags to `@type` declarations?